packages/react-native-lockdown/src/index.js

```diff
@@ -0,0 +1,200 @@
+ */
+
+/**
+ * @typedef {(options: { platform?: string }) => ReadonlyArray<string>} GetPolyfills
+ * @see {@link https://metrobundler.dev/docs/configuration/#getpolyfills}
+ */
+
+const { warn } = console
+const { assign } = Object
+
+const path = require('node:path')
+
+/**
+ * Default implementation of Metro serializer option 'getRunModuleStatement'.
+ *
+ * @type {GetRunModuleStatement}
+ */
+const defaultGetRunModuleStatement = (moduleId) => `__r(${moduleId})`
+
+const ENTRY_FILE_MODULE_ID = 0
+
+/**
+ * Creates a function to inspect for anomalies during module loading, then warns
+ * when complete.
+ *
+ * @returns {(callLog: (number | string)[]) => void} Function to inspect the
+ *   call log
+ */
+const warnAboutAnomalies = () => {
+  const deferredWarnings = {
+    calledOnceOnly:
+      'LavaMoat: getRunModuleStatement was only called once instead of at least twice in quick succession, with react-native InitializeCore first and bundle entry second. This is suspicious. Lockdown might not be in effect.',
+    neverCalledWithEntryModule:
+      'LavaMoat: getRunModuleStatement was not called with moduleId 0. Lockdown was not applied.',
+  }
+
+  // Emit warnings only after Metro finishes bundling
+  process.on('exit', () => {
+    Object.values(deferredWarnings).forEach((message) => {
+      if (message) warn(message)
+    })
+  })
+
+  /**
+   * Inspects the call log for anomalies.
+   *
+   * @param {(number | string)[]} callLog - Array of module IDs that were loaded
+   * @returns {void}
+   */
+  return function inspectCallLog(callLog) {
+    if (callLog.length > 1) {
+      deferredWarnings.calledOnceOnly = false
+    }
+
+    if (callLog.includes(ENTRY_FILE_MODULE_ID)) {
+      deferredWarnings.neverCalledWithEntryModule = false
+    }
+
+    if (callLog[0] === ENTRY_FILE_MODULE_ID) {
+      warn(
+        'LavaMoat: getRunModuleStatement was called with moduleId 0 first. That should represent the bundle entrypoint but react-native InitializeCore should run first. This is suspicious. Lockdown might not be in effect or might be breaking react-native core polyfills.'
+      )
+    }
+
+    if (new Set(callLog).size !== callLog.length) {
+      warn(
+        `LavaMoat: getRunModuleStatement was called with the same moduleId multiple times. It might be because some build is running twice with the same config instance. [${callLog.join(', ')}]`
+      )
+    }
+  }
+}
+
+/**
+ * @typedef {Object} LockdownSerializerOptions
+ * @property {boolean} [hermesRuntime=true] - Whether to enable Hermes runtime
+ *   support (otherwise JavaScript Core). Default is `true`
+ */
+
+/**
+ * Creates a Metro serializer configuration with Hardened JavaScript.
+ *
+ * @param {LockdownSerializerOptions} [options={}] - Configuration options.
+ *   Default is `{}`
+ * @param {SerializerConfig} [userConfig={}] - User-provided serializer
+ *   configuration. Default is `{}`
+ * @returns {SerializerConfig} Configured serializer options with LavaMoat
```

Comment:
```suggestion
 * @returns {Omit<SerializerConfig, 'getRunModuleStatement' | 'getPolyfills'> & {getRunModuleStatement: GetRunModuleStatement, getPolyfills: GetPolyfills}}
```

The return type is guaranteed to have these two properties. Since that's a mouthful, you might want to stuff it in a `@typedef` (e.g., `LockdownSerializerConfig`).

Reply:
`Partial<SerializerConfig>` could also make it less mouthfully, but your suggestion is more complete